shared-module/gamepad/__init__.c

```diff
@@ -59,24 +59,15 @@ void gamepad_tick(void) {
             bit <<= 1;
         }
         current ^= self->pulls;
-        self->pressed |= last & current;
-    } else if (MP_OBJ_IS_TYPE(MP_OBJ_FROM_PTR(singleton), &gamepadshift_type)) {
+        self->pressed |= self->last & current;
+        self->last = current;
+    }
+    #if CIRCUITPY_GAMEPADSHIFT
+    else if (MP_OBJ_IS_TYPE(MP_OBJ_FROM_PTR(singleton), &gamepadshift_type)) {
         // buttons connected to a shift register
-        gamepadshift_obj_t *self = singleton;
-
-        common_hal_digitalio_digitalinout_set_value(self->latch_pin, 1);
-        for (int i = 0; i < 8; ++i) {
-            common_hal_digitalio_digitalinout_set_value(self->clock_pin, 0);
-            if (common_hal_digitalio_digitalinout_get_value(self->data_pin)) {
-                current |= bit;
-            }
-            common_hal_digitalio_digitalinout_set_value(self->clock_pin, 1);
-            bit <<= 1;
-        }
-        common_hal_digitalio_digitalinout_set_value(self->latch_pin, 0);
-        self->pressed |= last & current;
+        gamepadshift_tick(singleton);
```

Comment:
I wonder if it would even make sense to completely separate this and call this from the system tick.

Reply:
Done. Everything is split now. Please take another look. Thanks!